ipa-core/src/protocol/dp/distributions.rs

```diff
@@ -63,44 +61,130 @@ impl Distribution<f64> for RoundedBoxMuller {
         self.inner.sample(rng).round()
     }
 }
-
 impl From<BoxMuller> for RoundedBoxMuller {
     fn from(value: BoxMuller) -> Self {
         Self { inner: value }
     }
 }
 
+///  What follows is the implementation of sampling from the Truncated Double Geometric distribution which in the
+///  process samples from Double Geometric and Geometric distributions.
+
+///  Geometric Distribution
+/// Generates a sample from a geometric distribution with the given success probability.
+
+#[derive(Debug, PartialEq)]
+pub struct Geometric {
+    bernoulli: Bernoulli,
+}
+impl Geometric {
```

Comment:
we seem to prefer empty lines before definitions - I don't know why formatter is not flagging this, but it would be good to follow to common style